pkg/plugins/trivy/image.go

```diff
@@ -677,20 +677,20 @@ func getCommandAndArgs(ctx trivyoperator.PluginContext, mode Mode, imageRef, tri
 	args = append(args,
 		"--output",
 		fmt.Sprintf("/tmp/scan/%s 2>/tmp/scan/%s.log", resultFileName, resultFileName),
+		buildTrailingCommandArgs(resultFileName, trivyOperatorConfig.CompressLogs()),
 	)
 
-	if trivyOperatorConfig.CompressLogs() {
-		// Add compress arguments.
-		args = append(args,
-			fmt.Sprintf(`&& bzip2 -c /tmp/scan/%s | base64`, resultFileName),
-		)
+	return []string{"/bin/sh"}, append([]string{"-c"}, strings.Join(args, " "))
```

Comment:
TBH I'm not a huge fan of running a shell to capture commands as it can be used in a malicious way to inject something. 


WDYT if we remove the reliance on arbitrary shell commands and capture the output (and the error codes) of Go library commands (such as compress logs, encode to base64, capture the output etc.?

Reply:
@simar7 sorry, it seems I misunderstood something

These commands are executed inside a separate k8s Job based on the Trivy image.
I didn’t fully understand your point about using Go code directly.

To be honest, I’m also not a big fan of reading the report and/or the logs as container logs, but this was done by design and I don’t have a better solution at the moment. tried using volumes, but they also come with their own overhead.

Reply:
I think we can take this opportunity to get rid of `getCommandAndArgs` completely and replace it with actual library calls. Since we build these args and eventually call trivy, we should really use trivy as a library rather than call trivy binary.